wurst/closures/ClosureKeyPresses.wurst

```diff
@@ -0,0 +1,106 @@
+*/
+
+
+/* CONSTANTS */
+
+constant KEY_TRIGGER = CreateTrigger()..addAction(function onEvent)
+constant KEY_MAP = new HashMap<int, KeyPressListener>
+    
+
+/* CLOSURES */
+
+public function onKeyPress(oskeytype key, KeyPressListener listener) returns KeyPressListener
+    return onKeyPress(key, OSKEY_META.NONE, true, listener)
+
+public function onKeyPress(oskeytype key, OSKEY_META metaKey, KeyPressListener listener) returns KeyPressListener
+    return onKeyPress(key, metaKey, true, listener)
+
+public function onKeyPress(oskeytype key,  bool onKeyDown, KeyPressListener listener) returns KeyPressListener
+    return onKeyPress(key, OSKEY_META.NONE, onKeyDown, listener)
+
+public function onKeyPress(oskeytype key, OSKEY_META metaKey, bool onKeyDown, KeyPressListener listener) returns KeyPressListener
+    for i=0 to bj_MAX_PLAYERS
+        if players[i].isIngame()
+            KEY_TRIGGER.registerPlayerKeyPress(players[i], key, metaKey, onKeyDown)
+    return registerListener(listener, key, metaKey, onKeyDown)
+
+public function onKeyPress(player whichPlayer, oskeytype key, KeyPressListener listener) returns KeyPressListener
+    return onKeyPress(whichPlayer, key, OSKEY_META.NONE, true, listener)
+
+public function onKeyPress(player whichPlayer, oskeytype key, OSKEY_META metaKey, KeyPressListener listener) returns KeyPressListener
+    return onKeyPress(whichPlayer, key, metaKey, true, listener)
+        
+public function onKeyPress(player whichPlayer, oskeytype key,  bool onKeyDown, KeyPressListener listener) returns KeyPressListener
+    return onKeyPress(whichPlayer, key, OSKEY_META.NONE, onKeyDown, listener)
+
+public function onKeyPress(player whichPlayer, oskeytype key, OSKEY_META metaKey, bool onKeyDown, KeyPressListener listener) returns KeyPressListener
+    KEY_TRIGGER.registerPlayerKeyPress(whichPlayer, key, metaKey, onKeyDown)
+    return registerListener(listener, key, metaKey, onKeyDown)
+
+function registerListener(KeyPressListener listener, oskeytype key, OSKEY_META metaKey, bool onKeyDown) returns KeyPressListener
+    listener.key = key
+    listener.metaKey = metaKey.toInt()
+    listener.onKeyDown = onKeyDown
+    let id = key.getHandleId()
+    let first = KEY_MAP.get(id)
+    if first != null 
+        first.prev = listener 
+        listener.next = first 
+    KEY_MAP.put(id, listener)
+    return listener 
+
+
+/* ON EVENT */
+
+function onEvent() 
+    var listener = KEY_MAP.get(BlzGetTriggerPlayerKey().getHandleId())
+    let metaKey = BlzGetTriggerPlayerMetaKey()
+    let onKeyDown = BlzGetTriggerPlayerIsKeyDown()
+    while listener != null 
+        if listener.metaKey == metaKey and listener.onKeyDown == onKeyDown
+            listener.onEvent()
+        listener = listener.next 
+
+
+/* KEY PRESSES LISTENER */
+
+abstract class KeyPressListener
+    oskeytype key 
+    int metaKey
+    bool onKeyDown
+    thistype next = null 
```

Comment:
we should probably make a module for this now that we use it at multiple occasions.
Basically like LinkedListModule but with custom add/get/remove etc.
Not in this PR tho ofc, just noting.